org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java

```diff
@@ -107,29 +108,26 @@
  *
  * @author igor
  */
+@Component(service = {IProjectConfigurationManager.class, IMavenProjectChangedListener.class,
```

Comment:
While I love defaults. I think the default is bad and always try to make it explicit for the following reasons:

- If a component does not implement any interface it is "immediate" by default, what means it is always started. Lets say after a while you add an interface (e.g. `IAdaptable`) the component auto-implements this but now becomes lazy by default, and is not activated anymore --> could be quite confusing
- Also if it implements interfaces and I add another one I might not want it to be published as a service
- If I remove an interface I might not notice that it was a service interface, if I list them I get complains from the IDE that the component does not implement all listed services

Because of this I think its better to list things explicitly, it does not save much time/effort here thanks to we can use class-objects.

Reply:
Agree on that. I already assumed the points you mentioned are the reason to make it explicit. Thanks for the clarification.